doc improvement: doc cleanup mesh dfu (#11209)
nrfconnect/sdk-nrf: doc/nrf/protocols/bt_mesh/ble_mesh_dfu_samples.rst

Bluetooth mesh Device Firmware Update (DFU)
###########################################

In the |NCS|, there is support for Device Firmware Update (DFU) over Bluetooth® mesh when performing firmware over-the-air (FOTA) updates for your Bluetooth mesh devices and applications.
Bluetooth mesh supports the distribution of firmware images across a mesh network.
The Bluetooth mesh DFU subsystem implements the Firmware update section of the Bluetooth Mesh Model Specification v1.1.

> The |NCS| also supports FOTA updates with point-to-point DFU over Bluetooth Low Energy using the MCUmgr subsystem and the Simple Management Protocol (SMP).

Author: Yes, merging these two pages into one is what the ultimate aim is.
I am not sure we get to do this perfectly during the time we have before this PR has to be merged.
But I can give it a go, and we can always keep improving it later as well.

Provisioning and configuring the devices
========================================

> After programming the samples onto the boards, they need to be provisioned into the same Bluetooth mesh network with an external provisioner device like described in sample documentation (:ref:`Provisioning the mesh DFU target device <ble_mesh_dfu_target_provisioning>` and :ref:`Provisioning the mesh DFU distributor device <ble_mesh_dfu_distributor_provisioning>`).

Reviewer: ```suggestion
After programming the samples onto the boards, they need to be provisioned into the same Bluetooth mesh network with an external provisioner device.
See the documentation for :ref:`provisioning the mesh DFU target device <ble_mesh_dfu_target_provisioning>` and :ref:`provisioning the mesh DFU distributor device <ble_mesh_dfu_distributor_provisioning>`  for how this is done.
```
It's such a long sentence, that breaking it into two feels better. Not sure this split is the best, though.

To achieve this, issue the ``mesh models dfu slot add`` shell command specifying size in bytes of the image that was uploaded to the Distributor.
Optionally, you can provide firmware ID, metadata and Unique Resource Identifier (URI) parameters that come with the image.

> For example, to allocate a slot for :ref:`ble_mesh_dfu_target` sample with image size of 241236 bytes, with firmware ID set to ``0200000000000000``, and metadata generated as described in :ref:`bluetooth_mesh_dfu_eval_md` section below, type the following command::

Reviewer: ```suggestion
For example, to allocate a slot for the :ref:`ble_mesh_dfu_target` sample with image size of 241236 bytes, with firmware ID set to ``0200000000000000``, and metadata generated as described in :ref:`bluetooth_mesh_dfu_eval_md` section below, type the following command::
```
